src/docfx/publish/PublishModelBuilder.cs

```diff
@@ -47,13 +50,20 @@ public void AddOrUpdate(FilePath file, JObject? metadata, string? outputPath)
             if (!publishItems.ContainsKey(sourceFile))
             {
                 _buildOutput.TryGetValue(sourceFile, out var buildOutput);
+                string? sourceUrl = null;
+                if (sourceFile.Origin != FileOrigin.Redirection)
```

Comment:
This check seems unnecessary.

Reply:
Or if you want to return `null` for redirection, move the logic inside `GetGitUrl` method so other places that calls this method gets the correct same behavior.